### PR TITLE
Update jquery

### LINKED ISSUE #32: Update JQuery and JQueryUI to latest released versions
Our web UI libraries, JQuery and JQueryUI are both out of date. Ensure that we're using the latest of both and any other copies are entirely removed from the codebase. Test extensively.

### PR DESCRIPTION
**Pull request by andrewd18**
### Summary of Changes
Updates JQuery and JQueryUI to their latest versions.

### Benefits
- Removes JQuery 1.6.2 library, bringing us from June 2011 to August 2023. 
- Removes custom JQuery UI 1.13.2, upgrades to 1.14.0.
- Replaces custom JQuery UI theme with official JQueryUI dark-hive theme files.
- Closes #32  

### Risks
- Broken UI. Mitigated by manual UI testing of all features and extensive reading of JS console/error logs.
- Lack of browser support. IE 11 is no longer supported. Only Chrome, Firefox, Safari & Edge.

### Testing
- Manual.
- Verified that all actions I could perform on `main` were possible on this branch. Lots of trial and error.
- Added the beginnings of front-end unit tests and a corresponding GitHub runner.

### Possible improvements to this PR
- Did not update merchant.html or soloDraft.html since I think we're planning to get rid of them. I could either get rid of them in this PR or update them to work, if necessary.

### Other notes
- No affiliations were assimilated during the creation of this PR, although the Borg icon did try.

### CODE REVIEW

**Comment by datanoh** [comment]
I don't see any bugs outright, but is the animation different now? When I play a card it's sliding into place horizontally.

File: gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js [hunk 1]
```diff
@@ -292,10 +292,10 @@ export default class GameAnimations {
                     $(cardDiv).css(
                         {
                             position:"absolute",
-                            left:(gameWidth / 2 - cardWidth / 4),
-                            top:gameHeight * (3 / 8),
-                            width:cardWidth / 2,
-                            height:cardHeight / 2,
+                            left: "" + (gameWidth / 2 - cardWidth / 4) + "px",
```
[the remark this answers is not in the document]
Comment (datanoh) [reply gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js:295]: It's supposed to show up large in the center and then shrink down diagonally to its target position.

https://github.com/user-attachments/assets/b721b8de-0c90-4362-9f63-a1646f115cc4

[the remark this answers is not in the document]
Comment (andrewd18) [reply gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js:297]: Ready for re-review.

[the remark this answers is not in the document]
Comment (datanoh) [reply gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js:297]: It's still sliding in horizontally on my end

[the remark this answers is not in the document]
Comment (datanoh) [reply gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js:297]: It's beautiful....

[the remark this answers is not in the document]
Comment (datanoh) [reply gemp-module/gemp-stccg-client/src/main/web/js/gemp-022/gameAnimations.js:297]: Looks good on my end, feel free to merge!

**Pull request merged**